src/ai/defaultai.cc

```diff
@@ -4029,6 +4031,15 @@ bool DefaultAI::check_economies() {
 				get_economy_observer((*j)->economy(Widelands::wwWORKER))->flags.push_back(*j);
 				// and erase from this economy's observer
 				j = fl.erase(j);
+				if (eco_size != economies.size()) {
+					// economies was modified, so obs_iter is invalid now.
```

Comment:
What would modify economies?
There is economies.erase() below, but there is return immediately after it

Reply:
`get_economy_observer()` may add an item to economies, if `(*j)->economy()` is not yet in there.

Reply:
You are right.
What about using lazy approach? Just return True... We are doing the same few lines below... It would be less code and AI's scheduler is fine with returning True, it will skip some other stuff and delay would be only minimal.

Reply:
Done. I wasn't sure about that return value.

Reply:
It is ok for me now